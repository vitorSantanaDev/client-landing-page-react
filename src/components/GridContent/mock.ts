export default {
  title: 'Title text example',
  content: `<p>Lorem ipsum dolor sit amet, <a href="#">consectetur adipiscing</a> elit. Sed ac lacus pretium urna malesuada dignissim. Sed risus tortor, sagittis eget sem ut, porta imperdiet magna. In fringilla nisi enim. Aenean mollis vehicula nulla, non vulputate eros pretium eget. Vivamus tincidunt accumsan massa non convallis. Aliquam molestie efficitur faucibus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum suscipit convallis mi, vitae ornare dolor hendrerit quis. Nam gravida orci posuere nisi imperdiet posuere. Phasellus quis blandit turpis, eu feugiat velit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer ac malesuada quam. Aliquam eu finibus massa.</p>

  <p>Donec pulvinar volutpat ante quis porttitor. In hac habitasse platea dictumst. Sed eu placerat turpis, vitae pulvinar odio. Pellentesque viverra posuere enim, eu dapibus quam congue sit amet. Sed iaculis eleifend scelerisque. Curabitur ut elit at sem cursus euismod. Praesent ornare, urna vitae sollicitudin tincidunt, sem odio accumsan velit, quis iaculis neque elit vel ante. Suspendisse rhoncus felis at arcu blandit facilisis. Cras euismod varius nibh, a cursus nisi. Integer laoreet posuere sem vel pulvinar. Etiam vel blandit libero. Praesent consectetur tellus et justo lobortis rhoncus.</p>
  `,
  background: false
}
