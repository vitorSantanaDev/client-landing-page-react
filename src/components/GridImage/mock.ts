export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      image: {
        altText: 'Teste 1',
        srcImage: 'https://source.unsplash.com/random/800x800?r=1'
      }
    },
    {
      image: {
        altText: 'Teste 2',
        srcImage: 'https://source.unsplash.com/random/800x800?r=2'
      }
    },
    {
      image: {
        altText: 'Teste 3',
        srcImage: 'https://source.unsplash.com/random/800x800?r=3'
      }
    },
    {
      image: {
        altText: 'Teste 4',
        srcImage: 'https://source.unsplash.com/random/800x800?r=4'
      }
    },
    {
      image: {
        altText: 'Teste 5',
        srcImage: 'https://source.unsplash.com/random/800x800?r=5'
      }
    },
    {
      image: {
        altText: 'Teste 6',
        srcImage: 'https://source.unsplash.com/random/800x800?r=6'
      }
    }
  ]
}
