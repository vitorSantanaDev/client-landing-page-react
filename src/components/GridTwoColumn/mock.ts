export default {
  title: 'Grid two column',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ad vero, accusantium eum exercitationem in ex quo veniam sed totam, quos ipsam at. Facilis a labore alias sed, natus dolor temporibus?',
  srcImage: 'assets/images/javascript.svg'
}
