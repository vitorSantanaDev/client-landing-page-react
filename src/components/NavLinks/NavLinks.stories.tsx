import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import NavLinks from '.'
import links from './mock'

export default {
  title: 'NavLinks',
  component: NavLinks,
  args: {
    links: links
  }
} as ComponentMeta<typeof NavLinks>

export const Default: ComponentStory<typeof NavLinks> = (args) => (
  <NavLinks {...args} />
)
