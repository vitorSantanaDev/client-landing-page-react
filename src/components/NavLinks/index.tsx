import React from 'react'

import MenuLink from '../MenuLink'
import * as S from './styles'

interface INavLinks {
  links: {
    children: React.ReactNode
    link: string
    newTab: boolean
  }[]
}

const NavLinks: React.FC<INavLinks> = ({ links }) => (
  <S.Wrapper aria-label="Main menu">
    {links?.map((link) => (
      <MenuLink key={link.link} {...link} />
    ))}
  </S.Wrapper>
)

export default NavLinks
