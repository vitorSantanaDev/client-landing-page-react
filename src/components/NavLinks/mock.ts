export default [
  { children: 'First Link', link: '#target_1', newTab: false },
  { children: 'Second Link', link: '#target_2', newTab: true },
  { children: 'Tertiary Link', link: '#target_3', newTab: false },
  { children: 'Last Link', link: '#target_4', newTab: true },
  { children: 'Test Link', link: '#target_5', newTab: false },
  { children: 'Contact Link', link: '#target_6', newTab: false },
  { children: 'About Link', link: '#target_7', newTab: true }
]
