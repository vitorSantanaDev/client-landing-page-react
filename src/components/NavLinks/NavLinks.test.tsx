import React from 'react'
import { screen } from '@testing-library/react'

import { renderTheme } from '../../styles/theme/render-theme'
import { theme } from '../../styles/theme/theme'

import NavLinks from '.'
import mock from './mock'
describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />)
    const links = screen.getAllByRole('link')
    expect(links).toHaveLength(mock.length)
  })
  it('should render the nav in responsive format when the media query is triggered', () => {
    renderTheme(<NavLinks links={mock} />)
    const nav = screen.getByText(/First Link/).parentElement
    expect(nav).toHaveStyleRule('flex-flow', 'column wrap', {
      media: theme.media.lteMedium
    })
  })
  it('should not render links', () => {
    renderTheme(<NavLinks links={[]} />)
    const links = screen.queryAllByRole('link')
    expect(links).toHaveLength(0)
  })
  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
