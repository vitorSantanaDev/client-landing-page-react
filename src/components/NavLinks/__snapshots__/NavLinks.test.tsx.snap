// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c1 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  padding: 1.6rem;
  color: #0A1128;
  font-size: 1.6rem;
}

.c1::after {
  content: '';
  width: 0%;
  height: 0.2rem;
  bottom: 0.77rem;
  left: 50%;
  position: absolute;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  background-color: #dc143c;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="#target_1"
    target="_self"
  >
    First Link
  </a>
  <a
    class="c1"
    href="#target_2"
    target="_blank"
  >
    Second Link
  </a>
  <a
    class="c1"
    href="#target_3"
    target="_self"
  >
    Tertiary Link
  </a>
  <a
    class="c1"
    href="#target_4"
    target="_blank"
  >
    Last Link
  </a>
  <a
    class="c1"
    href="#target_5"
    target="_self"
  >
    Test Link
  </a>
  <a
    class="c1"
    href="#target_6"
    target="_self"
  >
    Contact Link
  </a>
  <a
    class="c1"
    href="#target_7"
    target="_blank"
  >
    About Link
  </a>
</nav>
`;
